### PR TITLE
[CPU][DEBUG_CAPS] Transformations dump: added CPU opset serialization for XmlBin format

### PR DESCRIPTION
**Pull request by v-Golubev**
### Details
Current `TransformationDumper` `XmlBin` format logic don't use any custom opsets when serialization is called. This might lead to the inability to read the result IR using `read_model` API. For instance, IR which contain `type_relaxed` nodes can't be read if serialization was done without extension opsets because such nodes are serialized as ov::opset.
The suggested solution is to pass CPU extension opsets as a `custom_opsets` parameter to the serialization pass

### CODE REVIEW

**Comment by dmitry-gorokhov** [comment]
@v-Golubev  I am wondering about usage scenarion. When we need to read IR serialized after some part of trasnformations using OV API? My understanding was that the only use-cases is vizualisation of such IR using Netron for example.

**Comment by v-Golubev** [comment]
@dmitry-gorokhov my usage scenario is local validation.
For instance, I added a new feature (e.g. LP Transformation) and want to check that I don't bring any degradations. For these purposes I use some script which goes through all models in some model cache and collects IRs after LPT passes on my feature and master branches. After that, the script compares the IR pairs using graphComparator (which is usually used in Transformation tests). This scenario allows us to detect the models that must be additionally investigated (in terms of perf/accuracy) and, more importantly, models that don't have to be investigated.

However, graphComparator compares only ov::Models so we have to read serialized IRs using OV API. If it's impossible, custom comparison logic have to be used.
Hope my example is clear enough.

**Comment by dmitry-gorokhov** [comment]
> @dmitry-gorokhov my usage scenario is local validation. For instance, I add a new feature (e.g. LP Transformation) and want to check that I don't bring any degradations. For this purpose I use some script which goes through all models in some model cache and collects IRs after LPT on my feature and master branches. After that, the script compares the IR pairs using graphComparator (which is usually used in Transformation tests). This scenario allows us to detect the models that must be additionally investigated (in terms of perf/accuracy) and, more importantly, models that don't have to be investigated.
> 
> However, graphComparator compares only ov::Models so we have to read serialized IRs using OV API. If it's impossible, custom comparison logic have to be used (I would like to avoid this). Hope my example is clear enough.

Thanks, sounds reasonable for me. Lets then add a notive [here ](https://github.com/v-Golubev/openvino/blob/vg/debug_caps/cpu_opset_serialization/src/plugins/intel_cpu/src/docs/debug_caps_filters.md?plain=1#L28) that dumped IR can be read using corresponding OV API

**Comment by v-Golubev** [comment]
@dmitry-gorokhov the note is added. Thanks

**Review by dmitry-gorokhov — APPROVED** [review]

File: src/plugins/intel_cpu/src/utils/ngraph_transformation.hpp [hunk 1]
```diff
@@ -66,8 +67,12 @@ class TransformationDumper {
 
         ov::pass::Manager serializer;
 
-        if (config.dumpIR.format.filter[DebugCapsConfig::IrFormatFilter::XmlBin])
-            serializer.register_pass<ov::pass::Serialize>(pathAndName + ".xml", "");
+        if (config.dumpIR.format.filter[DebugCapsConfig::IrFormatFilter::XmlBin]) {
+            std::map<std::string, ngraph::OpSet> custom_opsets;
+            auto opset = std::make_shared<Extension>()->getOpSets();
+            custom_opsets.insert(std::begin(opset), std::end(opset));
```
[the remark this answers is not in the document]
Comment (v-Golubev) [reply src/plugins/intel_cpu/src/utils/ngraph_transformation.hpp]: You are right, we don't need this additional map actually. Corrected, thanks!

**Review by EgorDuplensky — APPROVED** [review]

**Pull request merged**